src/main/java/org/jsoup/parser/TokenQueue.java

```diff
@@ -346,13 +346,13 @@ public String consumeTagName() {
     }
     
     /**
-     * Consume a CSS element selector (tag name, but | instead of : for namespaces, to not conflict with :pseudo selects).
+     * Consume a CSS element selector (tag name, but | instead of : for namespaces (or *| for wildcard namespace), to not conflict with :pseudo selects).
      * 
      * @return tag name
      */
     public String consumeElementSelector() {
         int start = pos;
-        while (!isEmpty() && (matchesWord() || matchesAny('|', '_', '-')))
+        while (!isEmpty() && (matchesWord() || matchesAny("*|","|", "_", "-")))
             pos++;
```

Comment:
it technically works by coincidence, since the 2nd time around the "|" will also match, I was considering just adding the `'*'` as a character in the list, but did want to force the combined `*|` heuristic to match.  I'm not sure if something like: `while (!isEmpty() && (matchesWord() || matchesAny('|', '_', '-') || (matches("*|") && (pos++ > 0)) pos++;` is better, or just more confusing from a code readability standpoint.  Open to thoughts/suggestions @jhy